tools/create_gain_map_test.py

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/env python3
+# Copyright (c) the JPEG XL Project Authors. All rights reserved.
+#
+# Use of this source code is governed by a BSD-style
+# license that can be found in the LICENSE file.
+
+"""
+Create a minimal JPEG XL file with a gain map (jhgm box) for testing.
+
+This generates a valid JXL container with:
+- Container signature
+- Minimal codestream (jxlc box) with a tiny valid image
+- Gain map (jhgm box) with test data matching libjxl's test format
+"""
+
+import struct
+import sys
+
+def write_box(box_type, content):
+    """Write a JPEG XL container box."""
+    box_size = len(content) + 8  # 4 bytes size + 4 bytes type + content
+    return struct.pack('>I', box_size) + box_type + content
+
+def create_minimal_jxl_codestream():
+    """
+    Create a minimal valid JPEG XL naked codestream for a 1x1 pixel image.
+    This is a very simplified version just for testing purposes.
+    """
+    # JXL signature for naked codestream
+    sig = bytes([0xff, 0x0a])
+
+    # Extremely minimal bitstream for 1x1 image
+    # This is a simplified representation - a real encoder would be more complex
+    # For testing, we'll use a known-good minimal codestream
+    minimal_stream = sig + bytes([
```

Comment:
I don't think this is a conforming bitstream, is it?

If you want a small valid bitstream, the smallest I could come up with is here (12 bytes):
https://github.com/mathiasbynens/small/pull/125